src/buffer/out/textBuffer.cpp

```diff
@@ -2207,3 +2208,34 @@ HRESULT TextBuffer::Reflow(TextBuffer& oldBuffer,
 
     return hr;
 }
+
+// Method Description:
+// - Adds a hyperlink to our hyperlink table
+// Arguments:
+// - The hyperlink URI
+void TextBuffer::AddHyperlinkToMap(std::wstring uri) noexcept

```

Comment:
We are going to have to consider how we trim these from the table as they go out of scope especially since the ID map can only hold ~64,000 of them.

Reply:
To start with, we could at least clear the map when the buffer is cleared, e.g. after RIS and ED3 sequences, or whatever console APIs clear the buffer.

Reply:
Fixed now - we trim from the hyperlink table when hyperlink references are removed from the buffer